nemo/collections/nlp/parts/utils_funcs.py

```diff
@@ -31,7 +31,8 @@
 from nemo.utils import logging
 
 
-def dtype_from_precision(precision: Union[int, str], megatron_amp_O2: Optional[bool]) -> torch.dtype:
+def torch_dtype_from_precision(precision: Union[int, str], megatron_amp_O2: Optional[bool] = None) -> torch.dtype:
+    """ Mapping from PTL precision types to corresponding PyTorch parameter datatype."""
     if megatron_amp_O2 is not None and megatron_amp_O2 is False:
         return torch.float32
```

Comment:
I'm thinking that this function shouldn't have the megatron_amp_O2 arg because the torch dtype from precision is a well-defined mapping.

We should add additional functions for the other use cases. For example, param_dtype (while training) depends on the mixed precision recipe (O1 or O2).

Reply:
Here's the attributes that we're trying to set:

- torch_dtype
- dtype
- params_dtype
- autocast_dtype

Let's define them clearly and then set them according to the proper utility function.  If possible, let's do this in the base model constructor for all models.